cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -523,6 +523,49 @@ __inline__ __device__ std::pair<char const*, char const*> trim_whitespaces_quote
   return {skip_character(trim_begin, quotechar), skip_character(trim_end, quotechar).base()};
 }
 
+/**
+ * @brief Adjusts the range to ignore starting/trailing whitespace characters.
+ *
+ * @param[in] begin Pointer to the first character in the parsing range
+ * @param[in] end Pointer to the first character after the parsing range
+ *
+ * @return Trimmed range
+ */
+__inline__ __device__ std::pair<char const*, char const*> trim_whitespaces(char const* begin,
+                                                                           char const* end)
+{
+  auto not_whitespace = [] __device__(auto c) { return !is_whitespace(c); };
+
+  auto const trim_begin = thrust::find_if(thrust::seq, begin, end, not_whitespace);
+  auto const trim_end   = thrust::find_if(thrust::seq,
+                                        thrust::make_reverse_iterator(end),
+                                        thrust::make_reverse_iterator(trim_begin),
+                                        not_whitespace);
+
+  return {trim_begin, trim_end.base()};
+}
+
+/**
+ * @brief Adjusts the range to ignore starting/trailing quotation characters.
+ *
+ * @param begin Pointer to the first character in the parsing range
+ * @param end Pointer to the first character after the parsing range
+ * @param quotechar The character used to denote quotes; '\0' if none
+ *
+ * @return Trimmed range
+ */
+__inline__ __device__ std::pair<char const*, char const*> trim_quotes(char const* begin,
```

Comment:
Am I crazy, or is this function not used anywhere now? It looks like you have currently inlined the logic for determining `is_quoted` (and it's a one-line thrust call, so I'm fine with that), making this function obsolete.

Reply:
It's used in `cpp/include/cudf/io/detail/data_casting.cuh`.